polaris-react/src/components/PositionedOverlay/utilities/math.ts

```diff
@@ -90,28 +129,95 @@ export function calculateHorizontalPosition(
   containerRect: Rect,
   overlayMargins: Margins,
   preferredAlignment: PreferredAlignment,
+  _scrollableContainerRect: Rect,
+  preferredHorizontalPosition?: 'left' | 'right',
+  overlayMinWidth = 0,
 ) {
-  const maximum = containerRect.width - overlayRect.width;
-
-  if (preferredAlignment === 'left') {
-    return Math.min(
-      maximum,
-      Math.max(0, activatorRect.left - overlayMargins.horizontal),
-    );
-  } else if (preferredAlignment === 'right') {
-    const activatorRight =
-      containerRect.width - (activatorRect.left + activatorRect.width);
-
-    return Math.min(
-      maximum,
-      Math.max(0, activatorRight - overlayMargins.horizontal),
-    );
+  const maximumWidth = containerRect.width - overlayRect.width;
+  const activatorRight =
+    containerRect.width - (activatorRect.left + activatorRect.width);
+
+  const minimumSurroundingSpace = overlayMargins.horizontal
+    ? overlayMargins.horizontal
+    : 16;
```

Comment:
i think our tokens are `rem` strings so it would take some mathing and type casting. Unless we have somewhere in our tokens that's compatible? I couldn't find it but i'm not super familiar with that codebase

Reply:
Yeah maybe, I was thinking we could grab them directly, before they get converted but you'd still have to parseInt to remove `px` so it's still jank

Reply:
The overlayMargins passed into these functions are grabbed from the overlay's contents as it varies by component (for example Popover has set margins, HoverCard has none).

https://github.com/Shopify/polaris/blob/14f24f142c0b44f1905859448ae9bc1d9fa6e86e/polaris-react/src/components/PositionedOverlay/PositionedOverlay.tsx#L313-L317